src/mongocxx/write_concern.cpp

```diff
@@ -84,8 +90,15 @@ bool write_concern::journal() const {
     return libmongoc::write_concern_get_journal(_impl->write_concern_t);
 }
 
-std::int32_t write_concern::nodes() const {
-    return libmongoc::write_concern_get_w(_impl->write_concern_t);
+stdx::optional<std::int32_t> write_concern::nodes() const {
+    std::int32_t w = libmongoc::write_concern_get_w(_impl->write_concern_t);
+    return w >= 1 ? stdx::optional<std::int32_t>{w} : stdx::nullopt;
+}
+
+stdx::optional<write_concern::level> write_concern::acknowledge_level() const {
+    std::int32_t w = libmongoc::write_concern_get_w(_impl->write_concern_t);
+    return w < 1 ? stdx::optional<write_concern::level>{static_cast<write_concern::level>(w)}
```

Comment:
There is a small problem here. What if the value that comes back from the C driver is something that isn't in range for the enumeration? e.g. you upgrade your libmongoc.so to something newer, and we have some new special write concern value, but the libmongocxx doesn't know that value. Right now you would get a 'write_concern::level' enum value back that wasn't actually in range.